app/javascript/components/share/share-actions.js

```diff
@@ -1,11 +1,32 @@
 import { createAction } from 'redux-actions';
+import { createThunkAction } from 'utils/redux';
 
-export const setShareModal = createAction('setShareModal');
-export const setShareUrl = createAction('setShareUrl');
-export const setShareType = createAction('setShareType');
+import { getShortenUrl } from 'services/bitly';
+
+const setShare = createThunkAction('setShare', params => dispatch => {
+  const shareData = { ...params };
```

Comment:
You can spread all values you need as `const { embedSettings, embedUrl, ..... } = params.data`